src/class/eventManagement/incomer.class.ts

```diff
@@ -208,11 +282,56 @@ export class Incomer <
       }
     } as IncomerChannelMessages<T>["IncomerMessages"];
 
+    if (!this.dispatcherIsAlive) {
+      this.logger.info(this.standardLogFn(finalEvent)("Event Stored but not published"));
+
+      return;
+    }
+
     await this.incomerChannel.publish(finalEvent);
 
     this.logger.info(this.standardLogFn(finalEvent)("Published event"));
   }
 
+  private async updateTransactionsStateTimeout() {
+    try {
+      await setTimeout(this.abortRegistrationTime, undefined, { signal: kCancelTimeout.signal });
+      kCancelTask.abort();
+    }
+    catch {
+      // Ignore
+    }
+  }
+
+  private async updateTransactionsState() {
+    try {
+      const transactions = await this.incomerTransactionStore.getTransactions();
+
+      const toResolve: Promise<any>[] = [];
+      for (const [transactionId, transaction] of Object.entries(transactions)) {
+        if (!transaction.published) {
+          toResolve.push(
+            Promise.all([
+              this.incomerChannel.publish({
+                ...transaction,
+                redisMetadata: {
+                  ...transaction.redisMetadata,
+                  transactionId
+                }
+              }),
+              this.incomerTransactionStore.updateTransaction(transactionId, { ...transaction, published: true })
+            ])
+          );
```

Comment:
Pas sur que ce soit nécessaire le double Promise.all ?

```suggestion
          toResolve.push(...[
            this.incomerChannel.publish({ ...transaction, redisMetadata: { ...transaction.redisMetadata, transactionId } }),
            this.incomerTransactionStore.updateTransaction(transactionId, { ...transaction, published: true })
          ]);
```

Reply:
C'est pour éviter qu'un `publish` soit décorrélé de son `updateTransaction`